### ISSUE TITLE
yaml syntax highlighting of single-quoted strings incorrect

### ISSUE THREAD

**Issue by miiichael** [vigna/ne#111]
Within single-quote YAML strings, backslashes are not special. This means
ne (and joe) currently highlights this incorrectly:

   ![image](https://user-images.githubusercontent.com/4179353/234497169-85f48dbc-94c2-473b-8326-186a599dc2e4.png)

`'\'` (a single backslash) is not the same as `"\"` (an incomplete double-quoted
string). Only `'` is special inside single-quoted strings, by my understanding.

**Comment by utoddl** [comment]
You are correct. Please try the `syntax/yaml.jsf` file from the `yaml-strings` branch ([here](https://raw.githubusercontent.com/vigna/ne/yaml-strings/syntax/yaml.jsf)) and let us know what you think.

**Comment by miiichael** [comment]
Yes, much better.
![image](https://user-images.githubusercontent.com/4179353/236387017-129893fb-7e01-454b-9e0b-85957d3cba75.png)
